### PR TITLE
Array, passed to Objects.hash(), should be wrapped into Arrays.hashCode()

### PR DESCRIPTION
**Pull request by dk2k**
Array, passed to Objects.hash(), should be wrapped into Arrays.hashCode()

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/resync/ResyncReplicationRequest.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public boolean equals(final Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(trimAboveSeqNo, maxSeenAutoIdTimestampOnPrimary, operations);
+        return Objects.hash(trimAboveSeqNo, maxSeenAutoIdTimestampOnPrimary, Arrays.hashCode(operations));
```
Comment (sandeshkr419) [inline server/src/main/java/org/opensearch/action/resync/ResyncReplicationRequest.java:106]: We are increasing time complexity by doing this change for sure. Is there a specific reason for doing this?

Comment (dbwiddis) [reply server/src/main/java/org/opensearch/action/resync/ResyncReplicationRequest.java:106]: @sandeshkr419 the hash of `operations` here is the array object's hash, but does not hash the contents.  You could have a case of two different arrays containing the same elements, which would produce a different hashCode here but match the `.equals()` check which uses `Arrays.equals()`.  This is a good change.

**Review by dbwiddis — COMMENTED** [review]
Code change LGTM.

@dk2k please add a test to [ResyncReplicationRequestTests](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/action/resync/ResyncReplicationRequestTests.java) to validate that `equals()` and `hashCode()` are consistent for two different arrays with the same elements.   It should fail before this change and succeed after it.

**Comment by dk2k** [comment]
Sorry, this rebase mess happened again.

**Comment by dbwiddis** [comment]
> Sorry, this rebase mess happened again.

@dk2k It actually took me a few years to figure out how to do it.  A `rebase` is different than a `merge`.  Here's some simple steps.
1. On your local machine, switch to the main branch: `git checkout main`
2. Get the latest from main: `git pull upstream main`
3. Switch back to your pr branch: `git checkout dk_array_hashcode`
4. Type `git rebase main`

At this point you may get merge conflicts.  You can usually resolve them in your editor.  Add/stage each file you fix the conflicts in, and then type `git rebase --continue`.

5. Once you've got everything rebased, force push to your fork PR branch: `git push -f`.

**Comment by dblock** [comment]
> @dk2k It actually took me a few years to figure out how to do it. A `rebase` is different than a `merge`. Here's some simple steps.

Sometimes that rebase is nasty. Here's an even less sophisticated way to get out of this: https://code.dblock.org/2015/08/31/getting-out-of-your-first-git-mess.html.

**Pull request closed without merging**